kernel/src/utilities/helpers.rs

```diff
@@ -66,6 +66,24 @@ macro_rules! count_expressions {
     ($head:expr, $($tail:expr),* $(,)?) => (1usize + count_expressions!($($tail),*));
 }
 
+/// Executables must specify their stack size by using the `stack_size!` macro.
+/// It takes a single argument, the desired stack size in bytes. Example:
+/// ```
+/// stack_size!{0x400}
```

Comment:
```suggestion
/// stack_size!{0x1000}
```

It would be great if we could give _some_ kind of guidance here on how to pick a stack size :/. I know we don't have that in practice a the moment ... I imagine that to-date new boards have just copy/pasted the stack size from whatever nearest-board they templated off of.

_Ideally_ this could be something like `declare_kernel_stack!{<optional_size_parameter>}` with a reasonable default value...

The majority of boards seem to be either `0x1000` or `0x2000`:
```
$ git grep STACK_MEMORY | cut -d';' -f2 | cut -d ']' -f1 | sort | uniq -c
  17  0x1000
   1  0x1400
   4  0x1500
   1  0x1700
  21  0x2000
   1  0x8000
   4  0x900
```

At minimum, the example given here is probably too small, given that our smallest board is `0x900`

Reply:
> explicit without a default is a good thing

But isn't 0x400 your default if that's what the comment says? We should pick a number that's plausible (0x1000 seems fine) rather than something that will certainly fail. No need to trap new users.

Reply:
Countering my own comment, I guess this is just docs for the utility. Most users will probably copy-paste from another board file.

Reply:
> But isn't 0x400 your default if that's what the comment says? We should pick a number that's plausible (0x1000 seems fine) rather than something that will certainly fail. No need to trap new users.

I have no issue changing the comment. I just wanted a direct copy and paste so that no one could accuse me of writing and submitting a macro :)

Reply:
Oh, that makes sense. 😆